integration-tests/src/test/java/com/datastax/oss/driver/mapper/NullSavingStrategyIT.java

```diff
@@ -51,13 +53,14 @@ public class NullSavingStrategyIT {
 
   private static final CcmRule CCM_RULE = CcmRule.getInstance();
 
-  private static final SessionRule<CqlSession> SESSION_RULE =
-      SessionRule.builder(CCM_RULE)
-          .withConfigLoader(
-              DriverConfigLoader.programmaticBuilder()
-                  .withString(DefaultDriverOption.PROTOCOL_VERSION, "V3")
-                  .build())
-          .build();
+  private static final SessionRule<CqlSession> SESSION_RULE = SessionRule.builder(CCM_RULE).build();
+
+  // JAVA-3076: V3 protocol calls that could trigger cassandra to issue client warnings appear to be
+  // inherently unstable when used at the same time as V4+ protocol clients. For this test we'll use
+  // latest protocol version for SessionRule set-up (i.e. creating a keyspace which could result in
+  // warning about too many keyspaces) and then create a new client for the tests to use in a
+  // separate @BeforeClass method. This client is torn down in @AfterClass method #teardown().
+  private static CqlSession v3Session;
```

Comment:
I'm a bit confused as to why we're keeping this session around as a static reference when we really only need it for a one-off operation.

Reply:
We need to close this session at the end of the test class, for `SESSION_RULE` this is handled automatically by junit which calls `SessionRule#after()` (and subsequently `session.close()`) because it's a `@ClassRule`

Reply:
Maybe the comments could be clearer here - this session instance is being used in all the tests via the `mapper` object so we need to keep it around until after they've all finished